src/rpc/services/isForkDisputedService/IsForkDisputedRpcMethods.ts

```diff
@@ -0,0 +1,123 @@
+import ARpcMethods from "@/rpc/ARpcMethods";
+import { ATransport } from "@/transport";
+import IsForkDisputedService from "./IsForkDisputedService";
+import { ChannelId, ForkId } from "@/types/types";
+
+class IsForkDisputedRpcMethods extends ARpcMethods {
+    service: IsForkDisputedService;
+
+    constructor(transport: ATransport, service: IsForkDisputedService) {
+        super(transport, service.p2pManager);
+        this.service = service;
+    }
+
+    /**
+     * Peer receives dispute acknowledgment request
+     * Check if fork is disputed and respond accordingly
+     */
+    public async onDisputeAcknowledgmentRequest(
+        channelId: ChannelId,
+        forkId: ForkId
+    ) {
+        console.log(
+            `Received dispute acknowledgment request for fork ${forkId}`
+        );
+
+        // Check if we already sent a request for this forkId
+        if (
+            this.service.hasAcknowledgedDisputedFork(
+                this.senderTransport,
+                forkId
+            )
+        ) {
+            console.log(
+                `Already sent request for fork ${forkId}, disconnecting`
+            );
+            return this.p2pManager.disconnectAndBlacklistPeer(
+                this.senderTransport
+            );
+        }
+
+        // Check if fork is disputed locally
+        const isDisputedLocal =
+            await this.p2pManager.stateManager.diamondStateMachine.localDiamondContract.isForkDisputed(
+                channelId,
+                forkId
+            );
+
+        if (isDisputedLocal) {
+            console.log(
+                `Fork ${forkId} is disputed on local diamond, responding`
+            );
+            return this.respondToDisputeAcknowledgment(channelId, forkId);
+        }
+
+        // Check if fork is disputed on chain
+        const isDisputedOnChain =
+            await this.p2pManager.stateManager.stateChannelManagerContract.isForkDisputed(
+                channelId,
+                forkId
+            );
+
+        if (isDisputedOnChain) {
+            console.log(`Fork ${forkId} is disputed on-chain, responding`);
+            return this.respondToDisputeAcknowledgment(channelId, forkId);
+        }
+
+        // Fork is not disputed - disconnect
+        console.log(`Fork ${forkId} is not disputed, disconnecting`);
+        return this.p2pManager.disconnectAndBlacklistPeer(this.senderTransport);
+    }
+
+    /**
+     * Respond to dispute acknowledgment
+     */
+    private async respondToDisputeAcknowledgment(
```

Comment:
these helper functions should be in the service and only entry points should be exposed as public functions in `RpcMethods` - during runtime `typescript` is striped and a byzantine actor can invoke these private functions as rpc{serviceName,privateMethodName,args}